core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -911,6 +914,9 @@ object Parser extends ParserInstances {
       case _ => Impl.Map1(p, fn)
     }
 
+  def select[A, B](p: Parser[Either[A, B]])(fn: Parser[A => B]): Parser[B] =
```

Comment:
For bonus points we can add select10 that takes a Parser1 then a Parser and returns a Parser1. Similarly a select01 that does the Parser1 second.

If we had that, we can override filter on Parser1 to return a Parser1

Reply:
Actually with this formulation I think we only need select and select1. In select1 the first argument is a Parser1.